### DISCUSSION TITLE
How to create components on new page with parameters (value and correct order) of previews page?

### DISCUSSION THREAD

**Discussion by PhilipZi** [neuefische/questions#107] [Web]
Hallo Zusammen, 

die Frage fällt mir schwer, ich hoffe ihr könnt verstehen, was ich möchte. 
Ich würde gerne auf einer neuen Seite nach clicken des Play butten die konfigurierten Daten abbilden die ich zuvor gesetzt habe. 
hier ein Bild von den Daten die umgesetzt werden sollen:
![Screenshot 2022-08-24 at 11 58 44](https://user-images.githubusercontent.com/102238180/186390050-837423b3-d2b5-481c-8dfd-1bc755c02200.png)

ich habe versucht die Daten abzubilden mit folgendem Code: 
![Screenshot 2022-08-24 at 11 59 51](https://user-images.githubusercontent.com/102238180/186390274-4f312553-e7ef-4b61-997f-53f3e6b52275.png)

das Ergebnis ist das: 
![Screenshot 2022-08-24 at 12 00 37](https://user-images.githubusercontent.com/102238180/186390418-273d2b49-545b-412a-8307-d033e6e2c92d.png)

was mich hier stört, ist das es eigentlich nur eine Pause zwischen den exercisess abbilden soll und nicht noch eine nach der letzten exerciss ich weiß nicht wie ich das umsetzten kann, bin da etwas lost.
Außerdem kommt es zu komplikationen, wenn ich bei minuten nichts eintrage und wenn ich bei repetitions 0 als Anzahl angebe. Das würde ich gerne vermeiden. 

ich kann leider keinen PR zu diesem Fall mit angeben, da dies prototyped ist.
Wäre echt happy, wenn sich jemand diesem Fall annehmen kann und mir irgendwie helfen kann. Vielen Dank!

**Comment by JessicaLoers** [comment]
Hi Phillip,
ohne einen Link zum Code ist das Problem leider  schwer nachzuvollziehen. 

Arbeite bitte auf deinem aktuellen freature branch. 🙂

**Reply by PhilipZi** [reply]
würde ich sehr gerne, aber die code reviews behindern mich ein wenig daran, und wenn ich nicht hier dran arbeiten würde, wüsste ich nicht an was ich sonst arbeiten soll um voran zu kommen in meinem Projekt

**Reply by PhilipZi** [reply]
Hier nochmal der gesamte code:

```import Head from 'next/head';
import {useMemo} from 'react';

import ExerciseCard from '../components/ExerciseCard';
import PauseIcon from '../components/Icons/PauseIcon';
import MoveBackButton from '../components/MoveBackButton';
import StyledCards from '../components/StyledCards';
import useStore from '../hooks/useStore';

export default function CountDown() {
	const repetitionCounter = useStore(state => state.repetitionCounter);
	const pauseTimes = useStore(state => state.pauseTimes);

	const repetitions = repetitionCounter.length;
	const lastRepetition = repetitionCounter[repetitions - 1];
	const numElements = lastRepetition.repetitionCounter;

	const pauseTime = pauseTimes.length;
	const lastPause = pauseTimes[pauseTime - 1];

	const exercises = useMemo(() => {
		const _exercises = [];
		for (let i = 0; i < numElements; i++) {
			_exercises.push(() => (
				<div>
					<ExerciseCard />
					{lastPause.minutes > 0 || lastPause.seconds > 0 ? (
						<StyledCards variant="pause">
							<PauseIcon />
							<p>Pause</p>
							{lastPause ? (
								<a key={lastPause.id}>
									{lastPause.minutes.toString().padStart(2, '0')}:
									{lastPause?.seconds.toString().padStart(2, '0')}
								</a>
							) : (
								'00:00'
							)}
						</StyledCards>
					) : (
						''
					)}
				</div>
			));
		}
		return _exercises;
	}, [numElements, lastPause]);

	return (
		<>
			<Head>
				<title key="title">Count Down</title>
				<meta key="description" name="description" content="count down" />
			</Head>
			<MoveBackButton />
			<h1>hallo</h1>
			<div className="App">
				{exercises.map((Block, idx) => (
					<Block key={idx} />
				))}
			</div>
		</>
	);
}
```
